src/app/search/search-filters/sockets.tsx

```diff
@@ -180,24 +180,45 @@ const socketFilters: FilterDefinition[] = [
   {
     keywords: 'deepsight',
     description: tl('Filter.Deepsight'),
-    format: 'simple',
+    format: ['simple', 'query'],
     destinyVersion: 2,
-    filter: () => (item: DimItem) =>
-      Boolean(
-        item.bucket.inWeapons &&
-          item.sockets?.allSockets.find((s) => {
-            const plugDef = s.plugged?.plugDef;
-            return (
-              plugDef &&
-              // must be a deepsight resonance extractor plug
-              // to-do: use an PCH enum for this
-              plugDef.plug.plugCategoryIdentifier === 'crafting.plugs.weapons.mods.memories' &&
-              // but not the objectiveless stub used to blank out that socket
-              plugDef.objectives
-            );
-          })
-      ),
+    suggestions: ['complete', 'incomplete'],
+    filter:
+      ({ filterValue }) =>
+      (item: DimItem) => {
+        switch (filterValue) {
+          case 'deepsight':
+            return isDeepsight(item);
+          case 'complete':
+            return isDeepsight(item, true);
+          case 'incomplete':
+            return isDeepsight(item, false);
+        }
+      },
   },
 ];
 
 export default socketFilters;
+
+function isDeepsight(item: DimItem, checkComplete: Boolean | null = null) {
+  return Boolean(
+    item.bucket.inWeapons &&
+      item.sockets?.allSockets.find((s) => {
+        const plugDef = s.plugged?.plugDef;
+        const complete = // always true for null, checkcomplete XNOR complete
+          checkComplete === null
```

Comment:
```suggestion
          checkComplete === undefined
```
could skip the entire default null assignment above by just checking for the lack of a param here